livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/scenarios/matching/StepTemplate.kt

```diff
@@ -65,20 +65,18 @@ internal class StepTemplate(
             var isPreceededByVariable = false
             for ((i, c) in templateAsString.withIndex()) {
                 if (c == '{' && !isEscaped(templateAsString, i)) {
-                    if (isVariable) {
-                        throw IllegalFormatException("Illegal opening curly brace at position $i!\nOffending string was: $templateAsString")
-                    }
-                    if (isPreceededByVariable) {
-                        throw IllegalFormatException("Consecutive variables at position $i! StepTemplate must contain an intervening text fragment to keep them apart.\nOffending string was: $templateAsString")
-                    }
+                    validateVariables(isVariable, i, templateAsString, isPreceededByVariable)
                     isVariable = true
                     if (lastIndex < i) {
                         tokens.add(templateAsString.substring(lastIndex, i))
                     }
                     lastIndex = i
                 } else if (c == '}' && !isEscaped(templateAsString, i)) {
                     if (!isVariable) {
-                        throw IllegalFormatException("Illegal closing curly brace at position $i!\nOffending string was: $templateAsString")
+                        throw IllegalFormatException(
+                            "Illegal closing curly brace at position $i!\n" +
```

Comment:
might this fit in a single line now?

Reply:
My suggestion would be raw string with trim margin. reflects the message better.